comparer/comparer.go

```diff
@@ -35,9 +40,25 @@ type TestCase struct {
 
 // A Comparer allows comparing query results for test cases between a reference API and a test API.
 type Comparer struct {
-	RefAPI      PromAPI
-	TestAPI     PromAPI
-	QueryTweaks []*config.QueryTweak
+	refAPI         PromAPI
+	testAPI        PromAPI
+	queryTweaks    []*config.QueryTweak
+	compareOptions cmp.Options
+}
+
+// New returns a new Comparer.
+func New(refAPI, testAPI PromAPI, queryTweaks []*config.QueryTweak) *Comparer {
+	var options cmp.Options
+	addFloatCompareOptions(queryTweaks, &options)
+	addDropResultLabelsOptions(queryTweaks, &options)
+	queryTweaksCopy := make([]*config.QueryTweak, len(queryTweaks))
```

Comment:
Is this copy needed? We only ever just read from this, or not?

Reply:
Yes I can get rid of the defensive copy.